lib/src/cache/map_cache.dart

```diff
@@ -17,6 +17,8 @@ part of quiver.cache;
 /// A [Cache] that's backed by a [Map].
 class MapCache<K, V> implements Cache<K, V> {
   final Map<K, V> _map;
+  // Map of outstanding ifAbsent calls used to prevent concurrent loads of the same key.
+  final Map<K, FutureOr<V>> _outstanding = {};
```

Comment:
`final _outstanding = <K, FutureOr<V>>{}` otherwise `_outstanding` will be `Map<dynamic, dynamic>` at runtime.